tools/symbol_converter/main.py

```diff
@@ -0,0 +1,213 @@
+
+    # handle hidden pins
+    for pin in raw_pins.values():
+        if not pin["hide"]:
+            continue
+        match = [ppin for ppin in pins.values() if ppin["name"] == pin["name"]]
+        # assert (len(match) > 0), f"did not find matching pins for hidden pin {pin} in component {name}"
+        logger.warning(
+            f"Did not find matching pins for hidden pin {pin} in component {name}. Ignoring component"
+        )
+        return f"#Skipped invalid component {name}"
+        for ppin in match:
+            ppin["aliases"].append(pin["alt_number"])
+
+    faebryk_if_map = {}
+    unnamed_if_cnt = 0
+    for no, pin in pins.items():
+        if pin["name"] == "~":
+            faebryk_if_map[no] = f"_unnamed[{unnamed_if_cnt}]"
+            unnamed_if_cnt += 1
+        else:
+            pin_raw_name = pin["name"]
+            pin_name = sanitize_name(pin_raw_name)
+            if type(pin_name) is tuple and pin_name[0] is None:  # TODO use exception
+                logger.error(
+                    f"Unescapable pin name in pin {pin_raw_name}: [{pin_name[1]}] in symbol {name}"
+                )
+                return f"#Skipped invalid component {name}"
+
+            faebryk_if_map[no] = pin_name
+
+    ifs_exp = "\n        ".join(
+        [
+            f"self.IFs.{_if} = Electrical()"
+            for pinno, _if in faebryk_if_map.items()
+            if pins[pinno]["name"] != "~"
+        ]
+    )
+
+    pinmap = dict(faebryk_if_map)
+    for no, pin in faebryk_if_map.items():
+        for aliased_no in pins[no]["aliases"]:
+            pinmap[aliased_no] = pin
+
+    # raw
+    raw_symbol = pprint.pformat(symbol["_raw"], indent=4, width=88).replace(
+        "\n", "\n        "
+    )
+
+    #
+    traits = []
+
+    # footprint & footprint pinmap
+    footprint_str = symbol["properties"]["Footprint"].replace('"', "")
+    if footprint_str != "":
+        footprint_trait = 'has_kicad_manual_footprint("{}")'.format(footprint_str)
+        traits.append(footprint_trait)
+        pinmap_trait = "has_defined_footprint_pinmap({})".format(
+            "{"
+            + ", ".join([f"{pinno}: self.IFs.{_if}" for pinno, _if in pinmap.items()])
+            + "}"
+        )
+        traits.append(pinmap_trait)
+
+    #
+    traits.append(
+        'has_defined_kicad_ref("{}")'.format(symbol["properties"]["Reference"])
+    )
+
+    traits_exp = "\n        ".join([f"self.add_trait({trait})" for trait in traits])
+
+    template = f"""
+class {name}(Component):
+    \"\"\"
+    Generated by symbol_converter
+    {annotation}
+    source:
+        {raw_symbol} 
+    \"\"\"
+
+    def _setup_traits(self):
+        {traits_exp}
+        return
+
+    def _setup_interfaces(self):
+        self.IFs.add_all(times({unnamed_if_cnt}, Electrical))
+        {ifs_exp}
+
+    def __new__(cls, *args, **kwargs):
+        self = super().__new__(cls)
+        self._setup_traits()
```

Comment:
`self._setup_traits()` should be moved to under `self._setup_interfaces()` (line 170)